src/bonefish/trace/trace.hpp

```diff
@@ -54,14 +69,38 @@ inline const char* base_file_name(const char* file_path)
 
 #define BONEFISH_TRACE_NOARGS(fmt) \
     if (bonefish::trace::is_enabled()) { \
-        std::cerr << "[" << bonefish::trace::base_file_name(__FILE__) << ":" << __LINE__ << "][" << __FUNCTION__ << "] " \
-                << boost::format(fmt) << std::endl; \
+        std::ostringstream oss; \
+        oss << "[" << bonefish::trace::base_file_name(__FILE__) << ":" << __LINE__ << "][" << __FUNCTION__ << "] " \
+                << boost::format(fmt); \
+        bonefish::trace::trace(oss.str()); \
     }
 
 #define BONEFISH_TRACE_ARGS(fmt, ...) \
     if (bonefish::trace::is_enabled()) { \
-        std::cerr << "[" << bonefish::trace::base_file_name(__FILE__) << ":" << __LINE__ << "][" << __FUNCTION__ << "] " \
-                << (boost::format(fmt) % __VA_ARGS__) << std::endl; \
+        std::ostringstream oss; \
+        oss << "[" << bonefish::trace::base_file_name(__FILE__) << ":" << __LINE__ << "][" << __FUNCTION__ << "] " \
+                << (boost::format(fmt) % __VA_ARGS__); \
+        bonefish::trace::trace(oss.str()); \
+    }
+
+// Macro for facilitating error logging.
+#define BONEFISH_ERROR(...) \
+    BOOST_PP_IIF(BOOST_PP_GREATER(BOOST_PP_VARIADIC_SIZE(__VA_ARGS__), 1), BONEFISH_ERROR_ARGS(__VA_ARGS__), BONEFISH_ERROR_NOARGS(BOOST_PP_VARIADIC_ELEM(0, __VA_ARGS__)))
+
+#define BONEFISH_ERROR_NOARGS(fmt) \
+    if (bonefish::trace::is_enabled()) { \
```

Comment:
IMHO, ERROR msg shouldn't be related to trace

Reply:
Originally, bonefish::trace::enabled was a switch to turn on/off tracing subsystem (_trace_ here). So, to avoid a bunch of separate flags like _error_enabled, _trace_enabled and so on, bonefish::trace::enabled is reconsidered as a switch for a whole logging subsystem. Also minimizes the change.